express-mutations/server.ts

```diff
@@ -0,0 +1,114 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.post('/api/actors', async (req, res, next) => {
+  try {
+    const sql = `
+  insert into "actors" ("firstName", "lastName")
+  values ('Brendan', 'Eich')
+  returning *
+  `;
+    const results = await db.query(sql);
+    const post = results.rows;
+    if (!post) {
+      throw new ClientError(404, 'It is missing either firstName or lastName');
+    }
+    res.status(201).json(post);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.put('/api/actors/:actorId', async (req, res, next) => {
+  try {
+    const { firstName, lastName } = req.query;
+    if (lastName === undefined) {
+      throw new ClientError(400, 'lastName is undefined');
+    } else if (firstName === undefined) {
+      throw new ClientError(400, 'firstName is undefined');
+    }
+    const { actorId } = req.params;
+    if (actorId === undefined) {
+      throw new ClientError(400, 'ActorId is undefined');
+    }
+    const sql = `
+    update "actors"
+    set
+      "firstName" = $2,
+      "lastName" = $3
+    where "actorId" = $1
+    returning *
+    `;
+    const params = [actorId, firstName, lastName];
+    const results = await db.query(sql, params);
+    const update = results.rows;
+    if (!update) {
+      throw new ClientError(404, 'Could not load actorId');
```

Comment:
It would be nice to include the ID the request sent in the error message so they'll know the mistake they made.